client/packages.go

```diff
@@ -90,13 +90,16 @@ type ResultInfo struct {
 // FindOptions supports exactly one of the following options:
 // - Refresh: only return snaps that are refreshable
 // - Private: return snaps that are private
-// - Query: only return snaps that match the query string
+// - Query: only return snaps that match the query string.
+// The SearchNameOnly predicate can be used to specify that
+// a search action is restricted to the snaps' name fields only. By default
+// the search is performed on all of the snaps' description fields.
 type FindOptions struct {
-	Refresh bool
-	Private bool
-	Prefix  bool
-	Query   string
-	Section string
+	Refresh        bool
+	Private        bool
+	SearchNameOnly bool
```

Comment:
@niemeyer we would need to change Query too (not too descriptive) ... since prefix is used as a predicate for search the snap names only or all the snap fields, ... Either we keep that selection at the search term level (e.g. Name & Fields, but not too fond of that choice) or we add an extra field to filter the search (as Prefix was doing)